core/templates/services/html-length.service.ts

```diff
@@ -71,70 +78,67 @@ export class HtmlLengthService {
 
     if (customTags) {
       for (const customTag of customTags) {
         totalCharacters += this.getWeightForNonTextNodes(
-          customTag, 'character');
+          customTag, CALCULATION_TYPE_CHARACTER);
       }
     }
     return totalCharacters;
   }
 
   private calculateBaselineLength(
-      sanitizedHtml: string, wordOrCharacter: string): number {
+      sanitizedHtml: string, calculationType: CalculationType): number {
     let domparser = new DOMParser();
     let dom = domparser.parseFromString(sanitizedHtml, 'text/html');
     let totalWeight = 0;
     for (let tag of Array.from(dom.body.children)) {
       const ltag = tag.tagName.toLowerCase();
       if (this.textTags.includes(ltag)) {
         totalWeight +=
-        this.getWeightForTextNodes(tag as HTMLElement, wordOrCharacter);
+        this.getWeightForTextNodes(tag as HTMLElement, calculationType);
       }
     }
     return totalWeight;
   }
 
   private getWeightForTextNodes(
-      textNode: HTMLElement, wordOrCharacter: string): number {
+      textNode: HTMLElement, calculationType: CalculationType): number {
     const textContent = textNode.textContent || '';
 
-    if (wordOrCharacter === 'word') {
-      const words = textContent.trim().split(' ');
-      const wordCount = words.length;
-      return wordCount;
+    let textContentTrimmed = textContent.trim();
+    let textContentCount: number = 0;
+    if (calculationType === CALCULATION_TYPE_WORD) {
+      const words = textContentTrimmed.split(' ');
+      textContentCount = words.length;
+    } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
+      const characters = textContentTrimmed.split('\n').join('').split('');
+      textContentCount = characters.length;
     }
-    const characters = textContent.trim().split('\n').join('').split('');
-    let characterCount = characters.length;
-    return characterCount;
+    return textContentCount;
   }
 
   private getWeightForNonTextNodes(
-      nonTextNode: string, wordOrCharacter: string): number {
+      nonTextNode: string, calculationType: CalculationType): number {
     if (nonTextNode.includes('oppia-noninteractive-math')) {
       return 1;
     }
     const altTextMatch = nonTextNode
       .match(/alt-with-value="&amp;quot;([^&]*)&amp;quot;"/);
 
-    if (wordOrCharacter === 'word') {
-      let words = [];
-      if (altTextMatch && altTextMatch[1]) {
-        const altText = altTextMatch[1];
-        words = altText.trim().split(' ');
-      }
-      return words.length + 2; // +2 as a bonus for images with text.
-    }
-    let chars = [];
+    let text = [];
     if (altTextMatch && altTextMatch[1]) {
       const altText = altTextMatch[1];
-      chars = altText.trim().split(' ').join('').split('');
+      if (calculationType === CALCULATION_TYPE_WORD) {
+        text = altText.trim().split(' ');
+      } else if (calculationType === CALCULATION_TYPE_CHARACTER) {
+        text = altText.trim().split(' ').join('').split('');
```

Comment:
Why do you need to split, join, and split again? Can this be simplified?

Reply:
@seanlip Suppose we have a altText ```Hello Kids``` Now to get a character count we need to remove spaces in between words so that it doesn't get counted that's why we are splitting to create it ```[Hello, Kids]``` , then joining ```HelloKids``` to remove space and again splitting that into characters.

I found another way to write that is ```text = altText.trim().replace(/\s+/g, '').split('');``` should we ? It is working the same as above.

Reply:
Your revised way looks cleaner, but why do you need to split in order to find the length? Don't strings have length too?

I don't have a strong opinion on whether we need to count spaces or not for the character count. Let's go with the simpler option (probably yes).

Reply:
Now I have used space as a character, and before I was splitting to keep the word and character count identical.
Now it has also been solved in a simplified way.